apps/api/src/domain/models/digital-text/digital-text-page.entity.ts

```diff
@@ -0,0 +1,28 @@
+import { NonEmptyString } from '@coscrad/data-types';
+import { DTO } from '../../../types/DTO';
+import BaseDomainModel from '../BaseDomainModel';
+import { PageIdentifier } from './page-identifier';
+
+export default class DigitalTextPage extends BaseDomainModel {
+    @NonEmptyString({
+        label: 'identifier',
+        description: 'text identifier for the page',
+    })
+    readonly identifier: PageIdentifier;
+
+    constructor(dto: DTO<DigitalTextPage>) {
+        super();
+
+        if (!dto) return;
+
+        const { identifier } = dto;
+
+        this.identifier = identifier;
+    }
+
+    changeIdentifier(newIdentifier: PageIdentifier): DigitalTextPage {
```

Comment:
No. There is no way to do this yet. Until we need such functionality (and I'd prefer to soft delete the page and re-add it properly), we do not want to put the behaviour on the model. I know with the first few entities I added methods like this, but that was just to illustrate the possibilities. Now that we know the "rules of the game", let's only add such methods as we expose them via the command layer to users.

Reply:
Sounds good, I was wondering about that but kept it in because of the boilerplate...

Reply:
Yea- I see that I put it on the original toy book model. I'll remove this for now.